apps/order/schema.py

```diff
@@ -96,6 +98,79 @@ class Meta:
         filterset_class = OrderFilterSet
 
 
+class OrderTypeForStatistic(DjangoObjectType):
+    total_quantity = graphene.Int()
+
+    class Meta:
+        model = Order
+        fields = ('id', 'order_code', 'total_price')
+
+    def resolve_total_quantity(root, info, **kwargs):
+        return root.book_order.aggregate(Sum('quantity'))['quantity__sum']
+
+
+stat_to = datetime.date.today()
+stat_from = stat_from = stat_to - datetime.timedelta(30)
+
+
+class OrderStatType(graphene.ObjectType):
+    total_quantity = graphene.Int()
+    books_uploaded_count = graphene.Int()
+    orders_completed_count = graphene.Int()
+    books_ordered_count = graphene.Int()
+    stat = CustomDjangoListField(OrderTypeForStatistic)
+
+    class Meta:
+        fields = ()
+
+    def resolve_total_quantity(root, info, **kwargs):
+        return root.aggregate(Sum('book_order__quantity'))['book_order__quantity__sum']
+
+    @staticmethod
+    def resolve_books_uploaded_count(root, info, **kwargs):
+        '''
+        Returns total books uploaded count
+        '''
+        if info.context.user.user_type == User.UserType.ADMIN.value:
+            return Book.objects.count()
+        elif info.context.user.user_type == User.UserType.PUBLISHER.value:
+            return Book.objects.filter(publisher=info.context.user.publisher).count()
+        return Book.objects.filter(created_by=info.context.user).count()
+
+    @staticmethod
+    def resolve_orders_completed_count(root, info, **kwargs):
+        '''
+        Returns total orders completed in last 30 days
+        '''
+        return get_orders_qs(info).filter(
```

Comment:
why not use root here as well?

Reply:
@thenav56  Yes we can use, I used root .